app/forms/write_off_form.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class WriteOffForm < WasteCarriersEngine::BaseForm
+  attr_accessor :comment
+
+  validates :comment, presence: true, length: { maximum: 250 }
+
+  def submit(params, user)
+    # Assign the params for validation
+    self.comment = params[:comment]
+
+    return false unless valid?
+
+    ProcessWriteOffService.run(
+      finance_details: transient_registration.finance_details,
+      user: user,
+      comment: comment
+    )
+
+    true
```

Comment:
Should we drop this `true` and return whatever the `ProcessWriteOffService` does? Currently the service returns true or false depending on the success but we don't seem to do anything with that.

Reply:
I have thought about that, but it generates a dependency between this method/class and the service itself, which will impact negatively to maintainability

Reply:
I see where you are coming from. However it does mean that right now if the balance is 0, the service returns `false` and we don't do anything, but as far as the user knows they have successfully made a change. So I worry that could cause some confusion.

Reply:
I don't really worry about it, as the button for doing a refund won't show if the balance is 0 they can only trigger the feature hacking the URL somehow :D

Reply:
We do return `false` but is not meaningful for now, is there in case of future development on the feature :)

Reply:
OK, if they shouldn't be accessing it anyway that's alright with me for now.